Allow custom X startup file

loh-tar opened the pull request:
Some user missed that typical X startup files like .xinit or .Xresources
are not taken into account. Now can such users create their own startup
file in the system or user config directory and can do there what ever
seems appropriate.

The name must be "start-x" and it will be "sourced".
See Manual.txt for details.

Close #14

loh-tar committed:
Allow custom X startup file

Some user missed that typical X startup files like .xinit or .Xresources
are not taken into account. Now can such users create their own startup
file in the system or user config directory and can do there what ever
seems appropriate.

The name must be "start-x" and it will be "sourced".
See Manual.txt for details.

Close #14

loh-tar commented:
I hope this fit the needs, but I haven't it tested properly. Would be nice to become some feedback and real world examples.

Edit: BTW I was also inspired by this fork/patch 
4e53508eb793018a2ae854570b0acae01a1201fa

loh-tar committed:
Oops! Remove debug code

loh-tar committed:
Herewith we commit to never use a variable that starts with "own"

Well, sounds a little hammy but I guess we should have some security for
custom session startup scripts

jlkDE commented:
I just tested this patch - everything worked as documented for me, well done!
Of course wayland sessions don't profit from this feature, but that's to be expected.

loh-tar committed:
Don't "source" but call start-x file

In the past quite an effort was made not to simply "source" a (config) file.
So I feel better with this approach. Thisway there is also no need for
the "own-namespace-quirk". On the other hand can't we profit from tbsm's
info function

loh-tar commented:
Thanks for testing. Unfortunately, I have changed it now fundamentally. May you try it again?

Yes no Wayland, but that wasn't requested :-)
BTW I'm not sure if this whole patch is really needed, because I noticed that .xinit is used by startx. On the other hand I like my example with the log redirection

jlkDE commented:
Just for completeness: Tried again, still works!
